lib/manager/setup-cfg/extract.ts

```diff
@@ -0,0 +1,83 @@
+import { id as datasource } from '../../datasource/pypi';
+import pep440 from '../../versioning/pep440';
+import { PackageDependency, PackageFile, Result } from '../common';
+
+function getSectionName(str: string): string {
+  const [, sectionName] = /^\[\s*([^\s]+)\s*]\s*$/.exec(str) || [];
+  return sectionName;
+}
+
+function getSectionRecord(str: string): string {
+  const [, sectionRecord] = /^([^\s]+)\s+=/.exec(str) || [];
+  return sectionRecord;
+}
+
+function getDepType(section: string, record: string): null | string {
+  if (section === 'options') {
+    if (record === 'install_requires') {
+      return 'install';
+    }
+    if (record === 'setup_requires') {
+      return 'setup';
+    }
+    if (record === 'tests_require') {
+      return 'test';
+    }
+  }
+  return 'extra';
+}
+
+function parseDep(
+  line: string,
+  section: string,
+  record: string
+): PackageDependency | null {
+  const [, depName, currentValue] =
+    /\s+([-_a-zA-Z0-9]*)\s*(.*)/.exec(line) || [];
+  if (
+    section &&
+    record &&
+    depName &&
+    currentValue &&
+    pep440.isValid(currentValue)
+  ) {
+    const dep: PackageDependency = { datasource, depName, currentValue };
+    const depType = getDepType(section, record);
+    if (depType) {
+      dep.depType = depType;
+    }
+    return dep;
+  }
+  return null;
+}
+
+export function extractPackageFile(
+  content: string
+): Result<PackageFile | null> {
+  let sectionName = null;
+  let sectionRecord = null;
+
+  const deps: PackageDependency[] = [];
+  content
```

Comment:
Are you sure you don't want to use a .ini parser for this?

Reply:
I tried it at first, but for some reason it returned `undefined` for multiline `install_requres`